app/main.py

```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+def cache(func: callable) -> callable:
+    results = {}
+
+    def wrapper(*args, **kwargs) -> callable:
```

Comment:
You may consider using a decorator from `functools` module to the `wrapper` function as well.